model/modeldecoder/modeldecodertest/populator.go

```diff
@@ -119,14 +252,50 @@ func iterateStruct(v reflect.Value, key string, fn func(f reflect.Value, fKey st
 		}
 		fKey = fmt.Sprintf("%s%s", key, name)
 
-		if fTyp.Kind() == reflect.Struct {
+		// call the given function with every field
+		fn(f, fKey)
+		// check field type for recursive iteration
+		switch f.Kind() {
+		case reflect.Ptr:
+			if !f.IsZero() && fTyp.Elem().Kind() == reflect.Struct {
+				iterateStruct(f.Elem(), fKey, fn)
+			}
+		case reflect.Struct:
 			switch f.Interface().(type) {
-			case nullable.String, nullable.Int, nullable.Interface:
+			case nullable.String, nullable.Int, nullable.Bool, nullable.Float64,
+				nullable.Interface, nullable.HTTPHeader, nullable.TimeMicrosUnix:
 			default:
 				iterateStruct(f, fKey, fn)
 			}
+		case reflect.Map:
+			l := f.Len()
+			if l == 0 {
+				continue
+			}
```

Comment:
Since the map element type is the same for each value, check it once up front:

```suggestion
			if f.Type().Elem().Kind() != reflect.Struct {
				continue
			}
```